http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/Http2IT.java

```diff
@@ -0,0 +1,219 @@
+                }
+            }
+        }
+    }
+
+    private void request(AtomicInteger headerCount, VertxTestContext testContext) {
+
+        httpClient.request(HttpMethod.GET, 8443, baseUri.getHost(), BASE_ENDPOINT)
+                .compose(request -> {
+                    for (int i = 0; i < headerCount.get(); i++) {
+                        String headerName = "Header" + i;
+                        String headerValue = "Value" + i;
+                        request.putHeader(headerName, headerValue);
+                    }
+                    return request.send();
+                })
+                .onFailure(throwable -> {
+                    if (throwable instanceof Http2Exception &&
+                            throwable.getMessage().contains("Header size exceeded max allowed size")) {
+                        LOG.error("Expected error occurred: " + throwable.getMessage());
+                        testContext.completeNow();
+                    } else {
+                        LOG.error("Unexpected error: " + throwable.getMessage());
+                        testContext.failNow(throwable);
+                    }
+                })
+                .onSuccess(response -> {
+                    testContext.failNow(new AssertionError("Unexpected successful response"));
+                });
+    }
+
+    @Test
+    @DisplayName("Verify X-Content-Type-Options Header")
+    void verifyXContentTypeOptionsHeader(VertxTestContext vertxTestContext) {
+        Checkpoint checkpoint = vertxTestContext.checkpoint(2);
+        HttpClientOptions options = new HttpClientOptions()
+                .setSsl(true)
+                .setUseAlpn(true)
+                .setTrustAll(true)
+                .setVerifyHost(false)
+                .setProtocolVersion(HttpVersion.HTTP_2);
+        httpClient = vertx.createHttpClient(options);
+        httpClient.request(HttpMethod.GET, 8443, baseUri.getHost(), BASE_ENDPOINT)
+                .compose(request -> request.send()
+                        .compose(httpClientResponse -> {
+                            assertEquals(HttpStatus.SC_OK, httpClientResponse.statusCode());
+                            String contentTypeOptionsHeader = httpClientResponse.getHeader("X-Content-Type-Options");
+                            assertNotNull(contentTypeOptionsHeader);
+                            assertEquals("nosniff", contentTypeOptionsHeader);
+                            assertEquals(HttpVersion.HTTP_2, httpClientResponse.version());
+                            checkpoint.flag();
+                            return httpClientResponse.body();
+                        }))
+                .onSuccess(body -> {
+                    assertThat("Body response", body.toString().contains(GREETING));
+                    checkpoint.flag();
+                })
+                .onFailure(vertxTestContext::failNow);
+    }
+
+    @AfterAll
+    public static void closeConnections() {
+        httpClient.close();
```

Comment:
this class creates 5 HTTP clients and closes one.

Reply:
you are right I will correct it, good catch, thanks.